### PR TITLE
Automatically detect kind

### PR DESCRIPTION
**Pull request by malcolmholmes**
Currently, using actions that read files, the user needs to tell Grizzly whether a resource
includes an envelope or not, and if it doesn't include an envelope, the user must specify
a kind.

This PR adds in basic kind detection. It will first detect an envelope. If no envelope
present, it will ask each handler to detect its own resource type, hopefully resulting in
a kind that identifies that handler.

### CODE REVIEW

**Review by julienduchesne — COMMENTED** [review]
Where is the datasource detection coming from?

File: pkg/grizzly/parsing.go [hunk 1]
```diff
@@ -113,23 +116,29 @@ func ParseYAML(registry Registry, yamlFile string, opts Opts) (Resources, error)
...
 	var resources Resources
 	for i := 0; ; i++ {
+		var m map[string]any
 		err = decoder.Decode(&m)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
-			return nil, fmt.Errorf("Error decoding %s: %v", yamlFile, err)
+			return nil, fmt.Errorf("xError decoding %s: %v", yamlFile, err)
```
[the remark this answers is not in the document]
Comment (malcolmholmes) [reply pkg/grizzly/parsing.go:127]: oops, that snuck in. thanks.

**Comment by malcolmholmes** [comment]
> Where is the datasource detection coming from?

At present, only enveloped and dashboard detection are implemented. Datasource presumably would be simple too.

**Comment by julienduchesne** [comment]
> > Where is the datasource detection coming from?
> 
> At present, only enveloped and dashboard detection are implemented. Datasource presumably would be simple too.

There's tests for it though. What am I missing?

**Comment by malcolmholmes** [comment]
> There's tests for it though. What am I missing?

Yes, it either tests (a) that enveloped detection works on a kind that doesn't have its own detection yet or (b) without an envelope it throws an error.

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**